### ISSUE TITLE
Allow `InnerAssignment` for return statements inside multi-statement block

### ISSUE THREAD

**Issue by T1mL3arn** [adireddy/haxe-checkstyle#522]
I assumed `ignoreReturnAssignments = true` works for code like this:

```haxe
public var canPause(default, set):Bool = true;

function set_canPause(v:Bool):Bool {
  persistentUpdate = !v;
  return canPause = v;
}
```

but I get the warning. So it appeared to be designed this way - "return is only statement inside block". But this is kinda limited. Could it be another option to ignore return assignments inside multi-statement block?

**Comment by AlexHaxe** [comment]
you are correct, `ignoreReturnAssignments` only works in setters and with it being the only expression in its block.
I believe Java checkstyle doesn't even allow any such use case. 

checkstyle is a tool to find code smells and potential coding mistakes. that's why there are strict rules on what is an allowed inner assignment in a return expression. adding more exceptions and configuration parameters would help you save one line of code, but it makes checkstyle less useful since it might overlook cases that it should report. even with a config option that users need to enable manually, they might not be aware of all the exceptions that come with it.
